test/blackbox/common/DDSBlackboxTestsMonitorService.cpp

```diff
@@ -591,6 +676,72 @@ struct ConnectionListSampleValidator : public SampleValidator
+                            {
+                                if ((data.status_kind() == elem.status_kind()) &&
+                                (data.local_entity() == elem.local_entity()))
+                                {
+                                    //! Check for connections
+                                    bool expected_locators_found = true;
+                                    for (auto& incoming_connection : data.value().connection_list())
+                                    {
+                                        auto conn_it = std::find_if(elem.value().connection_list().begin(),
+                                        elem.value().connection_list().end(),
+                                        [&](const statistics::Connection&  connection)
+                                        {
+                                            if (connection.mode() == incoming_connection.mode())
+                                            {
+                                                //! check locators
+                                                bool same_locators = true;
+                                                for (size_t i = 0; i < connection.announced_locators().size(); i++)
+                                                {
+                                                    bool locator_found = false;
+                                                    for (size_t j = 0;
+                                                    j < incoming_connection.announced_locators().size(); j++)
+                                                    {
+                                                        //! provided comparison operator
+                                                        if (connection.announced_locators()[i] ==
+                                                        incoming_connection.announced_locators()[j])
+                                                        {
+                                                            locator_found = true;
+                                                        }
+                                                    }
+
+                                                    if (!locator_found)
+                                                    {
+                                                        same_locators = false;
+                                                        break;
+                                                    }
+                                                }
+
+                                                return same_locators;
+                                            }
+                                            else
+                                            {
+                                                return false;
+                                            }
+                                        });
+
+                                        if (conn_it != elem.value().connection_list().end())
```

Comment:
Maybe I'm not following the nested lambdas correctly but, if the find_if succeeds then doesn't that mean that the locators are the expected ones?

Reply:
Good catch ! the condition was just the opposite